### DISCUSSION TITLE
How to upload PDF file (and not only link)?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#839] [Q&A] [answered]
Hi

Mavo is able 
- to store url's in JSON
- images in GitHub

Is there a way to store PDF files?

I fried with:
<input property="pdf" name="file1" type="file" accept="application/pdf">

...but it only stores the url of the file, and not the file itself

Idea is to have :
1/ in edition mode a button to upload pdf file to storage 
2/ in read mode, the ability to download the file

Useful to put a resume in Github for example for students

**Comment by DmitrySharabin** [comment, marked as answer]
Will this help? https://codepen.io/dmitrysharabin/pen/gOrpyag

Please keep in mind a couple of things:
- I used version 0.2.3 of Mavo (https://get.mavo.io/0.2.3/mavo.js) to avoid the bug with clickable links in edit mode. Unfortunately, we still have this bug in the stable version.
- I used [mv-uploads](https://mavo.io/docs/storage#uploads) to illustrate that we can store data and files in separate storages. Just replace it with `mv-storage` to go with GitHub.
- In read mode, you won’t see any links until you really upload a file, and the corresponding property will get its value (a URL pointing to the file). Empty properties are not shown in read mode.

**Reply by LeaVerou** [reply]
FYI this link bug is at the top of my radar to fix after my spring teaching ends.
